src/sctools/platform.py

```diff
@@ -629,3 +629,304 @@ def attach_barcodes(cls, args=None):
+        # check that both the barcode length and position are given as arguments
+        # or that neither barcode length and position are given as arguments (XOR boolean logic)
+        cell_barcode_start_pos_exists = bool(args.cell_barcode_start_pos) or (args.cell_barcode_start_pos == 0)
+        cell_barcode_length_exists =  bool(args.cell_barcode_length)
+        if (cell_barcode_start_pos_exists != cell_barcode_length_exists):
+            argparse.ArgumentError('Invalid cell barocde position/length, both position and length must be provided by the user together')
+
+        molecule_barcode_start_pos_exists = bool(args.molecule_barcode_start_pos) or (args.molecule_barcode_start_pos == 0)
+        molecule_barcode_length_exists = bool(args.molecule_barcode_length)
+        if (molecule_barcode_start_pos_exists != molecule_barcode_length_exists):
+            argparse.ArgumentError('Invalid molecule barocde position/length, both position and length must be provided by the user together')
+
+
+        sample_barcode_start_pos_exists = bool(args.sample_barcode_start_pos) or (args.sample_barcode_start_pos == 0)
+        sample_barcode_length_exists = bool(args.sample_barcode_length)
+        if (sample_barcode_start_pos_exists != sample_barcode_length_exists):
+            argparse.ArgumentError('Invalid sample barocde position/length, both position and length must be provided by the user together')
+
+
+        # check that an index fastq is provided sample barcode length and position are given
+        if args.i1 is None and args.sample_barcode_length:
+            argparse.ArgumentError('An i7 index fastq file must be given to attach a sample barcode')
+
+        # check that cell and molecule barcodes don't overlap
+        if args.cell_barcode_length and args.molecule_barcode_length:
+            cls._validate_barcode_input(args.molecule_barcode_start_pos,
+                                        args.cell_barcode_start_pos + args.cell_barcode_length)
+
+        return args
+
+    @classmethod
+    def _validate_barcode_start_pos(cls, given_value):
+        """Validates that the barcode start position is greater than 0
+
+        Parameters
+        ----------
+        given_value : int
+            the given start position of the barcode to validate
+
+        Returns
+        -------
+        given_value : int
+            returns the start position if it is valid
+
+        """
+        return cls._validate_barcode_input(int(given_value), 0)
+
+    @classmethod
+    def _validate_barcode_length(cls, given_value):
+        """Validates that the barcode length  is greater than 1
+
+        Parameters
+        ----------
+        given_value : int
+            the given length of the barcode to validate
+
+        Returns
+        -------
+        given_value : int
+            returns the length if it is valid
+
+        """
+        return cls._validate_barcode_input(int(given_value), 1)
+
+    @classmethod
+    def _tag_bamfile(cls,
```

Comment:
Cool -- this didn't need to change. I think this means it should be refactored to add it to the superclass (`GenericPlatform`) and removed from `TenXV2`

Reply:
I was told to not alter the TenXV2 class until there is better test coverage.